third_party/k8s.io/kubernetes/hack/lib/golang.sh

```diff
@@ -0,0 +1,919 @@
+#!/usr/bin/env bash
```

Comment:
do we need all of these unrelated scripts?

Reply:
all these are "pulled in" (directly or not) by the verify-licenses script

Reply:
is it possible to build a simpler version of verify-licenses script that doesn't reference these scripts?

Reply:
as in, build something bespoke for gatekeeper?

I think my initial thesis was to take something "off the shelf", but I did end up patching the script a bit for us so I am not quite sure where to stand now.

It's nice to use the same tooling as k8s and it is built for the purposes of CNCF license compliance in particular, which is what we want here. I do also dislike, on the other hand, that I needed to copy like 5 scripts to get just that one.

Reply:
Again, I decided against `license-lint` while it was more "off the shelf"-y given the `allowed_repos` issue and some other papercuts (like an out of data underlying license checking library)

Reply:
if we did end up writing a bespoke script for CNCF license linting, abstracting away the language that it's written in, it could look like:

- setup, pull code
- use go-licenses to create a license view >> dump1
- download allowable licenses from spdx >> dump2
- check dump1 vs dump2
- reconcile any cncf exceptions

And that structure is basically the `verify-licenses` script. If folks feel strongly that it's better to go this route than to pull in all these extra files as dependencies, I am happy to write our own script. I'm somewhat on the fence here, hopefully this is a set it and forget it kind of thing.

Reply:
Is there a way we can call the k8s script instead of “forking” it? would be good if we can avoid having to maintain a fork.

Reply:
AFAIU, there's no way to do so as the script contains a few hard coded things & we needed a few tweaks:

- which repository to download (changed from k8s to g8r here)
- specifying an exception `rcrowley/go-metrics` for us needed to be added in
- specify the URL for the constraint framework license

---

Some of these changes could be contributed upstream and make the script more generic but not all may be and we would need to coordinate w k8s folks to have those changes accepted in the first place.

Reply:
@sozercan @ritazh what do y'all think about my answers above?

Reply:
as per our weekly mtg, I updated the script to remove any dependencies on other scripts and was able to take them all out 💯 thanks for the patience and all the suggestions.